pkg/loop/internal/relayer/pluginprovider/contractreader/contract_reader.go

```diff
@@ -121,6 +140,22 @@ func DecodeVersionedBytes(res any, vData *pb.VersionedBytes) error {
 			return fmt.Errorf("%w: %w", types.ErrInternal, err)
 		}
 		err = dec.Unmarshal(vData.Data, res)
+	case ValuesEncodingVersion:
+		protoValue := &valuespb.Value{}
+		err = proto.Unmarshal(vData.Data, protoValue)
+		if err != nil {
+			return fmt.Errorf("%w: %w", types.ErrInvalidType, err)
+		}
+		value, err2 := values.FromProto(protoValue)
+		if err2 != nil {
+			return fmt.Errorf("%w: %w", types.ErrInvalidType, err2)
+		}
+		valuePtr, ok := res.(*values.Value)
+		if !ok {
+			return fmt.Errorf("%w: result is not of Value type", types.ErrInvalidType)
+		}
+
+		*valuePtr = value
```

Comment:
Just unmarshal into the valuePtr.  If it's not a values.Value it will still work.

Reply:
done